core/common/src/main/java/alluxio/exception/AlluxioException.java

```diff
@@ -75,16 +75,67 @@ public AlluxioTException toAlluxioTException() {
    * @param e the {link AlluxioTException} to convert to a {@link AlluxioException}
    * @return a {@link AlluxioException} of the type specified in e, with the message specified in e
    */
+  @SuppressWarnings("unchecked")
```

Comment:
That's not very clean separation of the logic. I'd rather combine the two lines into one with an additional call of `getAlluxioExceptionClassfromString`